### ISSUE TITLE
Sales return enhancement?

### ISSUE THREAD

**Issue by odiea** [jekkos/opensourcepos#250]
At our location we have not allowed returns unless a valid receipt is given to us. I changed the following and added the else statement. This prevents the items list from showing in the dropdown box.The only issue is you must input the return first.
Is there a way that the takings view can be changed to show like the Detailed Sales Report view, that way if a customer comes in without their receipt but knows the approximate date of purchase we can find the item per that sale? Also can the item that was returned be removed from the sale it was returned from so that the receipt from their previous Sale no longer shows that item?

controllers/sales
function item_search()
	{
		$suggestions = array();

         if ($this->sale_lib->get_mode() == 'return') 
		{
			$this->sale_lib->is_valid_receipt($this->input->post('q')) && $suggestions[] = $this->input->post('q');
		}
		else
		{	
		$suggestions = array_merge($suggestions, $this->Item->get_item_search_suggestions($this->input->post('q'),$this->input->post('limit')));
		$suggestions = array_merge($suggestions, $this->Item_kit->get_item_kit_search_suggestions($this->input->post('q'),$this->input->post('limit')));
        }
		echo implode("\n", $suggestions);
	}

**Comment by SteveIreland** [comment]
Rather than removing the returned item from the original sale I'd rather just track the quantity returned against the original sale.  That way we don't impact the original sales dollars.  Then we would add code to insure that the quantity returned against that particular sale cannot be greater than what was originally ordered ... unless, of course, the store policy is to accept any return.

**Comment by daN4cat** [comment]
In the UK you cannot enforce the receipt only. If somebody shows you a transaction to you showing a bank or credit card account that's fine.
Moreover the introduction of stronger consumer protection laws you can return a defective item 6 months later or even some years later but for a partial refund.

So this rule cannot be valid for everybody.

**Comment by odiea** [comment]
Yes indeed a valid form of payment is also accepted. Our issue was not so good customers removing the tags or labels off of items then bringing them up to the counter and asking for a refund. In a thrift store no less.

**Comment by daN4cat** [comment]
Ok, but in that case people at the counter should ask for a proof of transaction.
